lib/mapObjects/CGHeroInstance.cpp

```diff
@@ -301,6 +307,7 @@ void CGHeroInstance::setHeroType(HeroTypeID heroType)
 
 void CGHeroInstance::initHero(CRandomGenerator & rand, const HeroTypeID & SUBID)
 {
+	blockVisit = true;
```

Comment:
this line already exists in constructor, so maybe problem is somewhere else

Reply:
Ahh yes, didn't mean to commit it